superchain/superchain.go

```diff
@@ -59,6 +66,34 @@ type ChainConfig struct {
 	// Chain is a simple string to identify the chain, within its superchain context.
 	// This matches the resource filename, it is not encoded in the config file itself.
 	Chain string `yaml:"-"`
+
+	// Hardfork Configuration Overrides
+	HardForkConfiguration `yaml:",inline"`
+}
+
+// setNilHardforkTimestampsToDefault overwrites each unspecified hardfork activation time override
+// with the superchain default.
+func (c *ChainConfig) setNilHardforkTimestampsToDefault(s *SuperchainConfig) {
```

Comment:
We might be able to simplify this if we iterate over the same structs like
```
    cVal := reflect.ValueOf(&c.HardForkConfiguration).Elem()
    sVal := reflect.ValueOf(&s.hardForkDefaults).Elem()

    for i := 0; i < cVal.NumField(); i++ {
        field := cVal.Field(i)
        if field.IsNil() {
            defaultValue := sVal.Field(i)
            field.Set(defaultValue)
        }
    }
}
```

Reply:
Embedded structs are still normal fields, so this should work.

Reply:
[8177e13](https://github.com/ethereum-optimism/superchain-registry/pull/101/commits/8177e13de677aa47fcc190faf7bca38998e7598b)